src/pathfinding_igvc/include/PathConstructor.h

```diff
@@ -13,7 +13,7 @@
 #include <OccupancyGridAdapter.h>
 
 class PathConstructor {
-    OccupancyGridAdapter* _occupancy_grid_adapter;
+    std::shared_ptr<OccupancyGridAdapter> _occupancy_grid_adapter;
```

Comment:
My first time using this, but it seems to fit this case since the function creating `PathConstructor` would be creating a pointer to a certain `OccupancyGridAdapter` instance, and copying the pointer to here. And the shared pointer would automatically get destroyed after all references to the instance are gone.

https://en.cppreference.com/w/cpp/memory/shared_ptr

```
The object is destroyed and its memory deallocated when either of the following happens:

the last remaining shared_ptr owning the object is destroyed;
the last remaining shared_ptr owning the object is assigned another pointer via operator= or reset().
```